src/product_logging/framework.rs

```diff
@@ -244,6 +244,151 @@ log4j.appender.FILE.layout=org.apache.log4j.xml.XMLLayout
+/// * `config` - The logging configuration for the container
+///
+/// # Example
+///
+/// ```
+/// use stackable_operator::{
+///     builder::{
+///         ConfigMapBuilder,
+///         meta::ObjectMetaBuilder,
+///     },
+///     config::fragment,
+///     product_logging,
+///     product_logging::spec::{
+///         ContainerLogConfig, ContainerLogConfigChoice, Logging,
+///     },
+/// };
+/// # use stackable_operator::product_logging::spec::default_logging;
+/// # use strum::{Display, EnumIter};
+/// #
+/// # #[derive(Clone, Display, Eq, EnumIter, Ord, PartialEq, PartialOrd)]
+/// # pub enum Container {
+/// #     MyProduct,
+/// # }
+/// #
+/// # let logging = fragment::validate::<Logging<Container>>(default_logging()).unwrap();
+///
+/// const STACKABLE_LOG_DIR: &str = "/stackable/log";
+/// const LOG4J2_CONFIG_FILE: &str = "log4j2.properties";
+/// const MY_PRODUCT_LOG_FILE: &str = "my-product.log4j2.xml";
+/// const MAX_LOG_FILE_SIZE_IN_MIB: u32 = 10;
+/// const CONSOLE_CONVERSION_PATTERN: &str = "%d{ISO8601} %-5p %m%n";
+///
+/// let mut cm_builder = ConfigMapBuilder::new();
+/// cm_builder.metadata(ObjectMetaBuilder::default().build());
+///
+/// if let Some(ContainerLogConfig {
+///     choice: Some(ContainerLogConfigChoice::Automatic(log_config)),
+/// }) = logging.containers.get(&Container::MyProduct)
+/// {
+///     cm_builder.add_data(
+///         LOG4J2_CONFIG_FILE,
+///         product_logging::framework::create_log4j2_config(
+///             &format!("{STACKABLE_LOG_DIR}/my-product"),
+///             MY_PRODUCT_LOG_FILE,
+///             MAX_LOG_FILE_SIZE_IN_MIB,
+///             CONSOLE_CONVERSION_PATTERN,
+///             log_config,
+///         ),
+///     );
+/// }
+///
+/// cm_builder.build().unwrap();
+/// ```
+pub fn create_log4j2_config(
+    log_dir: &str,
+    log_file: &str,
+    max_size_in_mib: u32,
+    console_conversion_pattern: &str,
+    config: &AutomaticContainerLogConfig,
+) -> String {
+    let number_of_archived_log_files = 1;
+
+    let logger_names = config
+        .loggers
+        .iter()
+        .filter(|(name, _)| name.as_str() != AutomaticContainerLogConfig::ROOT_LOGGER)
+        .map(|(name, _)| name.escape_default().to_string())
+        .collect::<Vec<String>>()
+        .join(", ");
+    let loggers = if logger_names.is_empty() {
+        "".to_string()
+    } else {
+        format!("loggers = {}", logger_names)
+    };
+    let logger_configs = config
+        .loggers
+        .iter()
+        .filter(|(name, _)| name.as_str() != AutomaticContainerLogConfig::ROOT_LOGGER)
+        .map(|(name, logger_config)| {
+            format!(
+                "logger.{name}.name = {name}\nlogger.{name}.level = {level}\n",
+                name = name.escape_default(),
+                level = logger_config.level.to_log4j_literal(),
+            )
+        })
+        .collect::<String>();
+
+    format!(
+        r#"
```

Comment:
**nit:** unnecessary newline

The resulting YAML looks like this:
```
  log4j.properties: |2-

    appenders = FILE, CONSOLE
```

Reply:
https://github.com/stackabletech/operator-rs/pull/540/commits/d88c30565464a017310770c7c623f02c24a3ad74